src/useTranslation.js

```diff
@@ -74,7 +74,7 @@ export function useTranslation(ns, props = {}) {
       if (bindI18nStore && i18n)
         bindI18nStore.split(' ').forEach(e => i18n.store.off(e, boundReset));
     };
-  }, []); // define props to trigger using effect on rerender (none here)
+  }, [namespaces.join()]); // re-run effect whenever list of namespaces changes
```

Comment:
I thought it might be safer not to rely on object identity in case people pass inlined arrays like `useTranslation(["x", "y"])` that are recreated on every render. That's why I'm turning the `namespaces` array into a string first.

Reply:
yes...saw that...think that's rather clever